pkg/infrastructure/azure/dns.go

```diff
@@ -0,0 +1,147 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/dns/armdns"
+	"github.com/openshift/installer/pkg/infrastructure/clusterapi"
+	"github.com/openshift/installer/pkg/types"
+)
+
+// Create DNS entries for azure.
+func createAzureDNSEntries(ctx context.Context, in clusterapi.InfraReadyInput) error {
+	clusterName := in.InfraID
+	private := in.InstallConfig.Config.Publish == types.InternalPublishingStrategy
+	resourceGroup := in.InstallConfig.Config.Azure.ResourceGroupName
```

Comment:
We use this resource group for creating records in the private zone, but need to use the `BaseDomainResourceGroupName` when creating records in the public zone.